lib/recorder.js

```diff
@@ -41,11 +42,18 @@ class ConsoleRecorder extends NoOpRecorder {
 }
 
 class HttpRecorder {
-    constructor(url, username, password, debug) {
-        this._options = {
+    constructor(url, username, password, debug, addHeaders) {
+        function auth(name, pass) {
```

Comment:
Do you have some usecases in mind where the application will need to send header parameters to APM server?